lib/shoulda/matchers/active_model/strict_validator.rb

```diff
@@ -0,0 +1,50 @@
+module Shoulda
+  module Matchers
+    module ActiveModel
+      module StrictValidator
+        def allow_description(allowed_values)
+          "doesn't raise when #{attribute} is set to #{allowed_values}"
+        end
+
+        def expected_message_from(attribute_message)
+          "#{human_attribute_name} #{attribute_message}"
+        end
+
+        def formatted_messages
+          [messages.first.message]
+        end
+
+        def messages_description
+          if has_messages?
+            ': ' + messages.first.message.inspect
+          else
+            ' no exception'
+          end
+        end
+
+        def expected_messages_description(expected_message)
+          if expected_message
+            "exception to include #{expected_message.inspect}"
+          else
+            "an exception to have been raised"
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.